source/common/router/config_impl.cc

```diff
@@ -490,7 +490,7 @@ bool RouteEntryImplBase::matchRoute(const Http::RequestHeaderMap& headers,
   matches &= Http::HeaderUtility::matchHeaders(headers, config_headers_);
   if (!config_query_parameters_.empty()) {
     Http::Utility::QueryParams query_parameters =
-        Http::Utility::parseQueryString(headers.getPathValue());
+        Http::Utility::parseAndDecodeQueryString(headers.getPathValue());
```

Comment:
I think we can make this the default, but I think we probably need a runtime flag to disable this for people who are matching values with percent encoding enabled.

Same applies to the other filters. I think it's ok to change the behavior in the admin handlers.

Reply:
@zuercher I'll revert the change to places outside admin handler for now and do runtime flagging in different PR. WDYT?